python/proton_cffi_builder.py

```diff
@@ -0,0 +1,283 @@
+    int            pn_message_set_correlation_id    (pn_message_t *msg, pn_atom_t id);
+    const char *   pn_message_get_content_type      (pn_message_t *msg);
+    int            pn_message_set_content_type      (pn_message_t *msg, const char *type);
+    const char *   pn_message_get_content_encoding  (pn_message_t *msg);
+    int            pn_message_set_content_encoding  (pn_message_t *msg, const char *encoding);
+    pn_timestamp_t pn_message_get_expiry_time       (pn_message_t *msg);
+    int            pn_message_set_expiry_time       (pn_message_t *msg, pn_timestamp_t time);
+    pn_timestamp_t pn_message_get_creation_time     (pn_message_t *msg);
+    int            pn_message_set_creation_time     (pn_message_t *msg, pn_timestamp_t time);
+    const char *   pn_message_get_group_id          (pn_message_t *msg);
+    int            pn_message_set_group_id          (pn_message_t *msg, const char *group_id);
+    pn_sequence_t  pn_message_get_group_sequence    (pn_message_t *msg);
+    int            pn_message_set_group_sequence    (pn_message_t *msg, pn_sequence_t n);
+    const char *   pn_message_get_reply_to_group_id (pn_message_t *msg);
+    int            pn_message_set_reply_to_group_id (pn_message_t *msg, const char *reply_to_group_id);
+    pn_data_t *pn_message_instructions(pn_message_t *msg);
+    pn_data_t *pn_message_annotations(pn_message_t *msg);
+    pn_data_t *pn_message_properties(pn_message_t *msg);
+    pn_data_t *pn_message_body(pn_message_t *msg);
+    int pn_message_decode(pn_message_t *msg, const char *bytes, size_t size);
+    int pn_message_encode(pn_message_t *msg, char *bytes, size_t *size);
+    ssize_t pn_message_encode2(pn_message_t *msg, pn_rwbytes_t *buf);
+    struct pn_link_t;
+    ssize_t pn_message_send(pn_message_t *msg, pn_link_t *sender, pn_rwbytes_t *buf);
+    int pn_message_data(pn_message_t *msg, pn_data_t *data);
+"""
+
+error_h = """
+    #define PROTON_ERROR_H ...
+
+    typedef struct pn_error_t pn_error_t;
+    
+    #define PN_OK ...
+    #define PN_EOS ...
+    #define PN_ERR ...
+    #define PN_OVERFLOW ...
+    #define PN_UNDERFLOW ...
+    #define PN_STATE_ERR ...
+    #define PN_ARG_ERR ...
+    #define PN_TIMEOUT ...
+    #define PN_INTR ...
+    #define PN_INPROGRESS ...
+    #define PN_OUT_OF_MEMORY ...
+    #define PN_ABORTED ...  
+
+    const char *pn_code(int code);
+    pn_error_t *pn_error(void);
+    void pn_error_free(pn_error_t *error);
+    void pn_error_clear(pn_error_t *error);
+    int pn_error_set(pn_error_t *error, int code, const char *text);
+//    int pn_error_vformat(pn_error_t *error, int code, const char *fmt, va_list ap);
+    int pn_error_format(pn_error_t *error, int code, const char *fmt, ...);
+    int pn_error_code(pn_error_t *error);
+
+    const char *pn_error_text(pn_error_t *error);
+
+    int pn_error_copy(pn_error_t *error, pn_error_t *src);
+
+
+"""
+
+
+ffi_builder.cdef(  type_h + object_h + error_h + codec_t +  message_h)
```

Comment:
you've mentioned you have a script to generate all the cdef strings. please add it to the PR. It does not matter if it's not yet completely finished, or completely polished.